### ISSUE TITLE
Modifying number of class

### ISSUE THREAD

**Issue by francis-bitonti** [nicolov/segmentation_keras#34]
Hi Nicolov,

Thanks for the great work! I tried to train new dataset by generating my own set of jpg and png masks. However I realized it only works for pre-defined 20 classes. For example I wanted to re-train this network to segment screws from background, I wasn't able to find way to add new classes but to use a existed color 0x181818 which was originally trained for cats. After training it did segmented the screw. However I'm still wondering is there any way to change the number of classes and specify which color value are associated with certain class?

**Comment by nicolov** [comment]
The only layer that cares about the number of classes is the last one. You can find the magic number [here](https://github.com/nicolov/segmentation_keras/blob/master/model.py#L46) (20 + 1 background)

**Comment by francis-bitonti** [comment]
Thanks a lot for the hint! Last question here, is it correct that the png greyscale value matches the numbers of classes? For example if the total class number is (5+1 background) the mask png pixel value would be 1~5 for different classes and 0 for background?

**Comment by nicolov** [comment]
Correct.

**Comment by DarylWM** [comment]
Hi Nicolov. Thanks from me also. To follow on from this question, would it be possible to use the pre-trained weights as long as in load_weights the weights and biases for the final layer were shrunk from shape (21,) into (6,) in @francisbitontistudio 's example?

**Comment by nicolov** [comment]
You would still need fine tuning of the last layer unless the 6 classes you are talking about are a subset of those used in the pretrained weights.

**Comment by DarylWM** [comment]
By "fine tuning the last layer" do you mean using train.py to train new
weights?

Daryl Wilding-McBride
Head of Technology



LinkedIn:  linkedin.com/company/dius-computing
Twitter:     @ <https://twitter.com/dius_au>DarylWMcB

**Comment by DarylWM** [comment]
By "fine tuning the last layer" do you mean using train.py to train new weights?

**Comment by nicolov** [comment]
something like this:
https://blog.keras.io/building-powerful-image-classification-models-using-very-little-data.html
